dragonfly/roompolygon.py

```diff
@@ -0,0 +1,421 @@
+        room_polyface = Polyface3D.from_offset_face(
+            self.geometry, self.floor_to_ceiling_height)
+        hb_room = Room.from_polyface3d(self.name, room_polyface)
+
+        # assign properties to walls
+        for i, bc in enumerate(self._boundary_conditions):
+            hb_room[i + 1]._boundary_condition = bc
+        if self._glazing_ratios is not None:
+            for i, (glz, glz_par) in enumerate(zip(self.glazing_ratios,
+                                                   self.glazing_parameters)):
+                if glz != 0:
+                    if glz_par is None:  # fast scaling operation to make Apertures
+                        face = hb_room[i + 1]
+                        scale_factor = glz ** .5
+                        ap_face = face.geometry.scale(scale_factor, face.geometry.center)
+                        aperture = Aperture(
+                            '{}_Glz1'.format(face.display_name), ap_face)
+                        aperture._parent = face
+                        face._apertures.append(aperture)
+                        # if the Aperture is interior, set adjacent boundary condition
+                        if isinstance(face._boundary_condition, Surface):
+                            names = face._boundary_condition.boundary_condition_objects
+                            adj_ap_name = '{}_Glz1'.format(names[0])
+                            final_names = (adj_ap_name,) + names
+                            aperture.boundary_condition = Surface(final_names, True)
+                    else:  # slower detailed operation to make Apertures
+                        hb_room[i + 1].apertures_by_ratio_rectangle(
+                            glz, glz_par.glazing_height, glz_par.sill_height,
+                            glz_par.horizontal_separation, glz_par.vertical_separation, 0)
+
+        # assign properties for the roof and floor from the parent Story
+        try:
+            hb_room[0].boundary_condition = bcs.adiabatic
+            hb_room[-1].boundary_condition = bcs.adiabatic
+        except AttributeError:
+            pass  # honeybee_energy is not loaded and Adiabatic type doesn't exist
+        if self._parent is not None:
+            if self._parent.is_ground_floor:
+                hb_room[0].boundary_condition = bcs.ground
+            if self._parent.is_top_floor:
+                hb_room[-1].boundary_condition = bcs.outdoors
+
+        return hb_room
+
+    @staticmethod
+    def solve_adjcency(room_polygons, tolerance):
+        """Solve for all adjacencies between a list of input RoomPolygons.
+
+        Args:
+            room_polygons: A list of RoomPolygons for which adjacencies will be solved.
+            tolerance: The minimum difference between the coordinate values of two
+                faces at which they can be considered centered adjacent.
+        """
+        for i, room_1 in enumerate(room_polygons):
+            try:
+                for room_2 in room_polygons[i + 1:]:
```

Comment:
Shouldn't you check the height of the rooms first and if the height is not the same then it will not be a match?

Reply:
I have been thinking about this and my answer for now is "No, not here". This is primarily because I know I want to support cases where you have adjacent Rooms with different floor-to-ceiling heights in the same Story, which is a very common situation. In other words, we definitely should not be keeping the boundary condition as outdoors or ground when this is the case because that's about the same as not supporting it.

For the first version of Dragonfly, I think I may make another version of this method that just sets all interior walls to be adiabatic and that's what we use for initial testing. Ultimately, though, I think I can get this to work even when the floors and/or ceilings are uneven since splitting an extrusion along its height is not a difficult task. But, in this case, all of the magic should be in the `to_honeybee()` method when we go from the 2D world over to the 3D one and the `solve_adjacency` method should be relatively unchanged from this version.

Reply:
I think it is easier not to support rooms with different heights until the details that you mentioned are implemented otherwise this can introduce bugs for the exact cases that mentioned above.

Reply:
I actually added full support for this since the last time I responded. As, I said, it was pretty easy to just split the wall surfaces along their height. I should note that, if someone has interior windows between adjacent walls, I enforce that the floor heights and floor-to-ceiling heights match. But, as long as there's no interior windows between adjacent segments, everything gets taken care of in the `to_honeybee()` method to ensure adjacent surfaces match.